Source/WebKitLegacy/win/WebHistoryItem.cpp

```diff
@@ -113,12 +115,8 @@ HRESULT WebHistoryItem::initFromDictionaryRepresentation(_In_opt_ void* dictiona
     bool lastVisitWasFailure = lastVisitWasFailureRef && CFBooleanGetValue(lastVisitWasFailureRef);
 
     std::unique_ptr<Vector<String>> redirectURLsVector;
-    if (CFArrayRef redirectURLsRef = static_cast<CFArrayRef>(CFDictionaryGetValue(dictionaryRef, redirectURLsKey))) {
-        CFIndex size = CFArrayGetCount(redirectURLsRef);
-        redirectURLsVector = makeUnique<Vector<String>>(size);
-        for (CFIndex i = 0; i < size; ++i)
-            (*redirectURLsVector)[i] = String(static_cast<CFStringRef>(CFArrayGetValueAtIndex(redirectURLsRef, i)));
-    }
+    if (CFArrayRef redirectURLsRef = dynamic_cf_cast<CFArrayRef>(CFDictionaryGetValue(dictionaryRef, redirectURLsKey)))
+        redirectURLsVector = WTFMove(makeVector<String>(redirectURLsRef));
```

Comment:
This `WTFMove` is not required, return values like this are already rvalue references, and I am surprised it compiles.

Reply:
The LHS is actually a `std::unique_ptr<Vector<String>>`, so I'm pretty sure `WTFMove()` was correct here.

Also, WinCairo doesn't build with the `WTFMove()` removed:  <https://ews-build.webkit.org/#/builders/12/builds/93305>

Reply:
Actually, I think I needed to use `makeUnique<Vector<String>>(WTFMove(makeVector<String>(redirectURLsRef)))` here.  Trying that now.

Reply:
I was wrong.  Removing `WTFMove()`:  <https://ews-build.webkit.org/#/builders/12/builds/93413>

```
C:\BW\WinCairo-EWS\build\WebKitBuild\Release\WTF\Headers\wtf/StdLibExtras.h(528): error C2338: T is not an lvalue reference; move() is unnecessary.
C:\BW\WinCairo-EWS\build\Source\WebKitLegacy\win\WebHistoryItem.cpp(118): note: see reference to function template instantiation 'WTF::Vector<WTF::String,0,WTF::CrashOnOverflow,16,WTF::VectorMalloc> &&std::move<WTF::CheckMoveParameter,WTF::Vector<WTF::String,0,WTF::CrashOnOverflow,16,WTF::VectorMalloc>>(T &&)' being compiled
        with
        [
            T=WTF::Vector<WTF::String,0,WTF::CrashOnOverflow,16,WTF::VectorMalloc>
        ]
```